huaweicloud/services/eg/data_source_huaweicloud_eg_official_event_channels.go

```diff
@@ -0,0 +1,132 @@
+				Optional: true,
+				Description: utils.SchemaDesc(
+					`The channel ID used to query specified official event channel.`,
+					utils.SchemaDescInput{
+						Deprecated: true,
+					},
+				),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The channel name used to query specified official event channel.`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `The ID of the enterprise project to which the official event channels belong.`,
+			},
+			"channels": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the official event channel.`,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The name of the official event channel.`,
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The description of the official event channel.`,
+						},
+						"provider_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The type of the official event channel.`,
+						},
+						"enterprise_project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The ID of the enterprise project to which the official event channel belongs.`,
+						},
+						"cross_account_ids": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: `The list of domain IDs (other tenants) for the cross-account policy.`,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"created_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The creation time of the official event channel.`,
+						},
+						"updated_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: `The latest update time of the official event channel.`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceOfficialEventChannelsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+	client, err := cfg.NewServiceClient("eg", region)
+	if err != nil {
+		return diag.Errorf("error creating EG client: %s", err)
+	}
+
+	channels, err := queryEventChannels(client, d, "OFFICIAL")
+	if err != nil {
+		return diag.Errorf("error querying official event channels: %s", err)
+	}
+
+	uuid, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(uuid)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("channels", flattenDataEventChannels(filterDataEventChannels(cfg, d, channels))),
```

Comment:
Method `flattenDataEventChannels` lacks a method to backfill **description**.

Reply:
yeah, description has been supported now